### ISSUE TITLE
PassHud does not show any records

### ISSUE THREAD

**Issue by stanbluijs** [mnussbaum/PassHUD#2]
### No Records are shown on opening PassHud

I installed pass (manually / without homebrew) in `/usr/local/password-manager` in the normal `bin`, `lib` folders. 
PassHud is installed via the dmg file verion `0.2.0`.

### What's the expected result?
I expect that on opening PassHud with the short key `cmd + /` to open and show any result on typing a record name.

### What's the actual result
PassHud does open and shows an empty application (screenshot 1)
When typing a record name no results are shown.

I opened PassHud in terminal and does not show any errors, but also no results (screenshot 2)

**Config file `~/.PassHUD`**
```
version: 1
pass:
  commandPath: /usr/local/password-store/bin/pass
```

Location of password store `~/.password-store`

If you need any more information, please let me know.

* screenshot 1
![screenshot-1](https://user-images.githubusercontent.com/7602123/52940076-f5cbd300-3365-11e9-8296-32b6c194384a.png)

* screenshot 2
![screenshot-2](https://user-images.githubusercontent.com/7602123/52940694-6c1d0500-3367-11e9-8f43-13e81bb9834c.png)

**Comment by rbuzatu90** [comment]
Hi @stanbluijs,

I had the same issue with the initial build and I fixed it by using absolute paths for tree and gpg and AFAIK also had an issue with getopt. Here's my fix, hope it works for you as well:
commit c2bfb8f964f8d35109bd12a4d9a80420026f1363
Author: Remus Buzatu <r.buzatu90@gmail.com>
Date:   Wed Jan 23 10:13:42 2019 +0400

    Used absolute path for commands for pass-hud compatibility
    
    Signed-off-by: Remus Buzatu <r.buzatu90@gmail.com>

diff --git a/pass-1.7.3 b/pass-1.7.3
index fe6bf58..7721421 100755
--- a/pass-1.7.3
+++ b/pass-1.7.3
@@ -7,10 +7,10 @@ umask "${PASSWORD_STORE_UMASK:-077}"
 set -o pipefail
 
 GPG_OPTS=( $PASSWORD_STORE_GPG_OPTS "--quiet" "--yes" "--compress-algo=none" "--no-encrypt-to" "--armor")
-GPG="gpg"
+GPG="/usr/local/bin/gpg"
 export GPG_TTY="${GPG_TTY:-$(tty 2>/dev/null)}"
-which gpg2 &>/dev/null && GPG="gpg2"
-[[ -n $GPG_AGENT_INFO || $GPG == "gpg2" ]] && GPG_OPTS+=( "--batch" "--use-agent" )
+which gpg2 &>/dev/null && GPG="/usr/local/bin/gpg"
+[[ -n $GPG_AGENT_INFO || $GPG == "/usr/local/bin/gpg" ]] && GPG_OPTS+=( "--batch" "--use-agent" )
 
 PASSWORD_STORE_CLIP_TIME="15"
 PASSWORD_STORE_ENABLE_EXTENSIONS=true
@@ -388,7 +388,7 @@ cmd_show() {
 		else
 			echo "${path%\/}"
 		fi
-		tree -C -l --noreport "$PREFIX/$path" | tail -n +2 | sed -E 's/\.gpg(\x1B\[[0-9]+m)?( ->|$)/\1\2/g' # remove .gpg at end of line, but keep colors
+		/usr/local/bin/tree -C -l --noreport "$PREFIX/$path" | tail -n +2 | sed -E 's/\.gpg(\x1B\[[0-9]+m)?( ->|$)/\1\2/g' # remove .gpg at end of line, but keep colors
 	elif [[ -z $path ]]; then
 		die "Error: password store is empty. Try \"pass init\"."
 	else
@@ -400,7 +400,7 @@ cmd_find() {
 	[[ $# -eq 0 ]] && die "Usage: $PROGRAM $COMMAND pass-names..."
 	IFS="," eval 'echo "Search Terms: $*"'
 	local terms="*$(printf '%s*|*' "$@")"
-	tree -C -l --noreport -P "${terms%|*}" --prune --matchdirs --ignore-case "$PREFIX" | tail -n +2 | sed -E 's/\.gpg(\x1B\[[0-9]+m)?( ->|$)/\1\2/g'
+	/usr/local/bin/tree -C -l --noreport -P "${terms%|*}" --prune --matchdirs --ignore-case "$PREFIX" | tail -n +2 | sed -E 's/\.gpg(\x1B\[[0-9]+m)?( ->|$)/\1\2/g'
 }
 
 cmd_grep() {

**Comment by stanbluijs** [comment]
Hi @rbuzatu90,

Thank you for your response.

I tried your fix. I added the paths for 'gpg', 'tree' and 'getopt'.

My .password-store looks like below
```
Password Store
|-- Company
|   |-- record-A
|   `-- record-B
`-- SomeFolder
    `-- record-C
`-- somerecord
```

In passHUD I only see 'Company' and 'SomeFolder' and the folder-less record but I cannot look into the folders. The folder-less record works like I expect. 

<img width="682" alt="passhud" src="https://user-images.githubusercontent.com/7602123/53122923-28d5b880-3558-11e9-8011-d4c5c2e6e281.png">

Does PassHud understand folders and should it know how to open/read them? Or could it be a problem with tree or something?

**Comment by mnussbaum** [comment]
Glad you've gotten it partially working! I'm curious if you set the `PATH` env var in `~/.PassHUD` if you still need the absolute paths in pass itself. Something like:

```yaml
---
version: 1

pass:
  commandPath: /usr/local/password-store/bin/pass
  env:
    - name: PATH
      value: /usr/local/bin
```

Or instead of setting just `/usr/local/bin` as the `PATH` in the config can you make sure it matches `PATH` in your terminal and includes `/usr/local/bin`?

Regarding the entries in folders, that's a separate issue, right now PassHUD isn't parsing entries in folders correctly. I might be able to fix this whenever I have a free moment but PRs are most welcome :).

**Comment by stanbluijs** [comment]
Hi,

I tried, with the original version of pass, the config you suggested, but this won't work.

```
$ cat ~/.PassHUD 
version: 1
pass:
  commandPath: /usr/local/password-store/bin/pass
  env:
    - name: PATH
      value: /usr/local/bin
```

Maybe I will look into the folder problem, but spare time is rare.

**Comment by mnussbaum** [comment]
I've updated PassHUD to use `/usr/libexec/path_helper` to find the system `PATH` configuration, which I think should fix the original issue and remove the need to patch pass itself with absolute paths.

I've released a version 0.2.1 with the fix. I'll open a new issue for the mis-handled directory problem

**Comment by mnussbaum** [comment]
@stanbluijs I've just released [version 0.2.2](https://github.com/mnussbaum/PassHUD/releases/tag/0.2.2) which should make PassHUD handle directories correctly. Please let me know if it works for you!

**Comment by stanbluijs** [comment]
@mnussbaum Thank you for the fix, I had the project (0.2.0) running and was trying to figure out how it works. First time with Swift for me :)
0.2.2 works like expected, thank you!
